cppcache/src/ClientMetadataService.cpp

```diff
@@ -578,12 +579,10 @@ ClientMetadataService::pruneNodes(
     const auto locations = metadata->adviseServerLocations(bucketId);
     if (locations.size() == 0) {
       LOGDEBUG(
-          "ClientMetadataService::pruneNodes Since no server location "
-          "available for bucketId = %d  putting it into "
-          "bucketSetWithoutServer ",
+          "ClientMetadataService::pruneNodes Use non single-hop path "
+          "since no server location is available for bucketId = %d",
           bucketId);
-      bucketSetWithoutServer.insert(bucketId);
-      continue;
+      return nullptr;
```

Comment:
I'd like to see an integration test that exposes the issue being fixed...

Reply:
Thank you for the review!

There is already integration "FunctionExecutionTest.FunctionExecutionWithIncompleteBucketLocations" test in this PR that exposes the issue. Please check comment https://github.com/apache/geode-native/pull/864#discussion_r705952887 .

Reply:
Hi @echobravopapa ,

Is this integration test OK to you? Maybe to remove request change if there is nothing else to add?

Thanks/Jakov